### PR TITLE
Adds 'only' to ObjectSchema to return a subset of properties

### PR DESCRIPTION
**Pull request by jillesme**
Implements my idea @ https://github.com/fastify/fluent-schema/issues/84

#### Checklist

- [x] run `npm run test` and `npm run benchmark` (benchmark wasn't defined)
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added * will do after review 
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: src/ObjectSchema.js [hunk 1]
```diff
@@ -278,6 +278,24 @@ const ObjectSchema = ({ schema = initialState, ...options } = {}) => {
...
     },
 
+    /**
+     * Returns an object schema with only a subset of keys provided
+     *
+     * @param properties a list of properties you want to keep
+     * @returns {ObjectSchema}
+     */
+    only: properties => {
+      return ObjectSchema({
```
Comment (jillesme) [inline src/ObjectSchema.js:288]:
```suggestion
    only: properties => {
      properties = (typeof properties === 'string') ? [properties] : properties;
      return ObjectSchema({
```

I was thinking to allow for a single string or a list of properties but decided that just an array of properties may be simpler. Happy to add ^ if y'all prefer `S.object().only('single_field')`

Comment (aboutlo) [reply src/ObjectSchema.js:288]: Array is good for me.

File: src/ObjectSchema.test.js [hunk 2]
```diff
@@ -804,6 +804,31 @@ describe('ObjectSchema', () => {
     })
   })
 
+  describe('only', () => {
+    it('returns a subset of the object', () => {
+      const base = S.object()
```
Comment (aboutlo) [inline src/ObjectSchema.test.js:809]: Not sure it's really needed but can you add a nested object too. Just to check we don't miss anything.

Comment (aboutlo) [reply src/ObjectSchema.test.js:809]: Can you add in the README an example of this use case? 
... and thank you for the nice addition! 🙏

**Comment by jillesme** [comment]
@aboutlo thanks for your fast review and feedback! I've reflected those changes in my last commit 😊

**Comment by jillesme** [comment]
@mcollina / @aboutlo - could this be merged at some point? Or if any changes are needed, please do let me know! ❤️

**Comment by aboutlo** [comment]
hey @jillesme javascript wise we are fine.  I just realized we missed Typescript definitions:
https://github.com/fastify/fluent-schema/blob/master/src/types/index.ts
https://github.com/fastify/fluent-schema/blob/master/src/FluentSchema.d.ts

Could add this as well? 🙏

**Comment by jillesme** [comment]
@aboutlo done!

**Pull request merged**